src/hotspot/share/interpreter/zero/bytecodeInterpreter.cpp

```diff
@@ -102,11 +102,13 @@
   There really shouldn't be any handles remaining to trash but this is cheap
   in relation to a safepoint.
 */
-#define SAFEPOINT                                                                 \
-    {                                                                             \
-       /* zap freed handles rather than GC'ing them */                            \
-       HandleMarkCleaner __hmc(THREAD);                                           \
-       CALL_VM(SafepointMechanism::process_if_requested(THREAD), handle_exception); \
+#define SAFEPOINT                                                      \
+    {                                                                  \
+       /* zap freed handles rather than GC'ing them */                 \
+       HandleMarkCleaner __hmc(THREAD);                                \
+       if (SafepointMechanism::should_process(THREAD)) {               \
+         CALL_VM({ThreadInVMfromJava tiv(THREAD);}, handle_exception); \
```

Comment:
I initially thought this was an okay convenience technique to get the desired effects, but seeing it used elsewhere I no longer think that - see comments "below".

Reply:
In this case I thought using TIVFJ seemed better because when reading CALL_VM() one would expect a transition from _thread_in_Java to _thread_in_vm and then a call to the appropiate method (like with JRT_ENTRY). It's just that in this case the call that we want to make (SafepointMechanism::process_if_requested()) is already included in the TIVFJ. Anyways, doing it the other way is okay too.